urllib3/contrib/gaepool.py

```diff
@@ -0,0 +1,101 @@
+import warnings
+
+from ..connectionpool import (
+    ConnectionPool,
+    _Default,
+    get_host)
+
+from ..exceptions import (
+    HTTPWarning,
+    TimeoutError,
+)
+
+from ..packages.six import BytesIO
+from ..request import RequestMethods
+from ..response import HTTPResponse
+from ..util.timeout import Timeout
+from ..util.url import Url
+
+try:
+    from google.appengine.api import urlfetch
+except ImportError:
+    urlfetch = None
+
+
+class GAEPlatformWarning(HTTPWarning):
+    pass
+
+
+class GAEConnectionPool(ConnectionPool, RequestMethods):
+
+    def __init__(self, host, port=None, scheme='http', timeout=5, headers=None,
+                 retries=None, validate_certificate=False):
+
+        if not urlfetch:
+            raise EnvironmentError(
+                "URLFetch is not available in this environment.")
+
+        ConnectionPool.__init__(self, host, port)
+        RequestMethods.__init__(self, headers)
+
+        if isinstance(timeout, Timeout):
+            if not timeout.read is timeout.connect is Timeout.DEFAULT_TIMEOUT:
+                warnings.warn(
+                    "URLFetch does not support granular timeout settings, "
+                    "reverting to total timeout.", GAEPlatformWarning)
+            timeout = timeout.total
+
+        self.deadline = timeout
+
+        self.scheme = scheme
+        self.validate_certificate = validate_certificate
+
+    def urlopen(self, method, url, body=None, headers=None, retries=None,
+                redirect=True, assert_same_host=True, timeout=_Default,
+                pool_timeout=None, release_conn=None, **response_kw):
```

Comment:
When you port this to a Manager, this stuff gets simplified. We don't care about `assert_same_host`, for example. See: https://github.com/shazow/urllib3/blob/master/urllib3/poolmanager.py#L153